### PR TITLE
feat: complete PISP side of linking flow

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

File: src/handlers/outbound/linking/request-consent/{ID}/authenticate.ts [hunk 1]
```diff
@@ -65,8 +65,13 @@ async function patch (_context: any, request: Request, h: StateResponseToolkit):
...
   } catch(error) {
-    // TODO: PUT /consents/{ID}/error to DFSP if PISP is unable to handle
-    //       the incoming POST /consents request
+    // todo: PUT /consentsRequest/{ID}/error to DFSP if PISP is unable to handle
+    //       the previous PUT /consentRequests/{ID} request
+    //       The handler doesn't know the DFSP's ID due to it being stored in the model
+    //       if the model is not found then we don't know the ID
+    //       We might need to pass the ID in LinkingRequestConsentIDAuthenticateRequest.
+    //       Though...do we need to notify the DFSP here...? Shouldn't it just be
+    //       the PISP? I don't think we do.
```
Comment (eoln) [inline src/handlers/outbound/linking/request-consent/{ID}/authenticate.ts:74]: authorizations errors have to go back to DFSP - DFSPTransactionModel, so DFSP will be able to change the TransactionRequestState to REJECTED and not allow to start simpleTransfer.

File: src/handlers/outbound/linking/request-consent/{ID}/pass-credential.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import * as OutboundAPI from '~/interface/outbound/api_interfaces'
+
+
+/**
+ * Handles outbound POST /linking/request-consent/{ID}/pass-credential request
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+// todo: consider changing this to PUT /linking/request-consent/{ID}/pass-credential
+//       since the flow triggers a PUT /consents/{ID}
+async function post (_context: any, request: Request, h: StateResponseToolkit): Promise<ResponseObject> {
```
Comment (eoln) [inline src/handlers/outbound/linking/request-consent/{ID}/pass-credential.ts:45]: we can drop using `any` and linter exception if we will use 'unknown' instead

File: src/handlers/outbound/linking/request-consent/{ID}/pass-credential.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      h.getLogger().error('outbound POST /linking/request-consent/{ID}/pass-credential unexpected result from workflow')
+      return h.response({}).code(500)
+    }
+
+    const statusCode = (result.currentState == 'errored') ? 500 : 200
+    return h.response(result).code(statusCode)
+  } catch(error) {
+    // todo: PUT /consents/{ID}/error to DFSP if PISP is unable to handle
+    //       the previous inbound POST /consents request
+    //       Do we need to notify the DFSP here...? Shouldn't it just be
```
Comment (eoln) [inline src/handlers/outbound/linking/request-consent/{ID}/pass-credential.ts:73]: we should notify DFSP here (DFSP model is awaiting on PUT or PUTError

File: src/handlers/outbound/linking/request-consent/{ID}/authenticate.ts (hunk 1 above)
Comment (eoln) [inline src/handlers/outbound/linking/request-consent/{ID}/authenticate.ts:74]: we should notify DFSP which is awaiting on PATCH or PUTError

**Review by eoln — APPROVED** [review]

File: src/handlers/outbound/linking/request-consent/{ID}/pass-credential.ts (hunk 3 above)
Comment (kleyow) [reply src/handlers/outbound/linking/request-consent/{ID}/pass-credential.ts:73]: ok.

File: src/handlers/outbound/linking/request-consent/{ID}/authenticate.ts (hunk 1 above)
Comment (kleyow) [reply src/handlers/outbound/linking/request-consent/{ID}/authenticate.ts:74]: ok.

File: src/handlers/outbound/linking/request-consent/{ID}/pass-credential.ts (hunk 2 above)
Comment (kleyow) [reply src/handlers/outbound/linking/request-consent/{ID}/pass-credential.ts:45]: I'll get in a follow up. copy/paste sry.

File: src/handlers/outbound/linking/request-consent/{ID}/authenticate.ts (hunk 1 above)
Comment (kleyow) [reply src/handlers/outbound/linking/request-consent/{ID}/authenticate.ts:74]: ok. i'll get this a follow up.

**Pull request merged**